tests/cuda_test/test_kernel_semantic.py

```diff
@@ -0,0 +1,218 @@
+
+# pylint: disable=missing-function-docstring, missing-module-docstring/
+import pytest
+
+import numpy as np
+from pyccel.epyccel import epyccel
+from pyccel.decorators import stack_array, types, kernel
+from pyccel.errors.errors import Errors, PyccelSemanticError
+from pyccel.errors.messages import (INVALID_FUNCTION_CALL, KERNEL_STACK_ARRAY_ARG, MISSING_KERNEL_CONFIGURATION,
+                                    NON_KERNEL_FUNCTION_CUDA_VAR,
+                                    INVALID_KERNEL_CALL_BP_GRID,
+                                    INVALID_KERNEL_CALL_TP_BLOCK
+                                    )
+
+@pytest.mark.parametrize( 'language', [
+        pytest.param("ccuda", marks = pytest.mark.ccuda)
+    ]
+)
+def test_stack_array_kernel(language):
+    @stack_array('arr')
+    def kernel_caller():
+        from numpy import ones
+        @kernel
+        @types('int[:]')
+        def stack_array_kernel(arr):
+            return arr[0]
+        arr = ones(1, dtype=int)
+        return stack_array_kernel[1,1](arr)
+
+    # Initialize singleton that stores Pyccel errors
+    errors = Errors()
+
+    # epyccel should raise an Exception
+    with pytest.raises(PyccelSemanticError):
+        epyccel(kernel_caller, language=language)
+
+    # Check that we got exactly 1 Pyccel error
+    assert errors.has_errors()
+    assert errors.num_messages() == 1
+
+    # Check that the error is correct
+    error_info = [*errors.error_info_map.values()][0][0]
+    assert error_info.symbol.func  == 'stack_array_kernel'
+    assert KERNEL_STACK_ARRAY_ARG == error_info.message
```

Comment:
Could this test ever be run with a different language? If not then I don't see any reason to pass this as an argument